lib/array_equals.rb

```diff
@@ -2,4 +2,51 @@
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
   raise NotImplementedError
+  if array1 == nil || array2 == nil
```

Comment:
When I think about how to write effective if-statements, I normally think of the flow like a triangle, where the stricter conditions should be going first, and the less strict conditions follow. The reason is if what you want to check does not meet the stricter conditions, you can quickly return the result without wasting time moving through the rest of the less strict conditions.